fedimint-core/src/task/jit.rs

```diff
@@ -88,24 +87,26 @@ where
         Fut: Future<Output = std::result::Result<T, E>> + 'static + MaybeSend,
     {
         let (cancel_tx, mut cancel_rx) = tokio::sync::mpsc::channel(1);
-        let (val_tx, val_rx) = oneshot::channel();
         let type_name = std::any::type_name::<T>();
+        let shared = Arc::new(JitCoreShared {
+            val: std::sync::OnceLock::new(),
+            initialized_waiter: Waiter::new(),
+        });
+        let shared2 = shared.clone();
         super::imp::spawn(
             &format!("JitTry {} value", std::any::type_name::<T>()),
             async move {
                 select! {
+                    biased;
```

Comment:
I doubt it matters, but if anything - I would rather get the value than cancelation if both finished at the same time.

Reply:
there is no point in getting the value when there is no one left to use it :)

Reply:
also non biased uses `rand` to decide which one to poll